### ISSUE TITLE
get_all_launch_configurations  result only names, not details

### ISSUE THREAD

**Issue by fvant** [boto/boto#3300]
This call returns only a list of lc names and not the details  (consequently, this causes issues in the Ansible scripts).

<code>aws autoscaling describe-launch-configurations</code>  does return the full details.

boto.Version = '2.38.0'

<code>import boto.ec2.autoscale
con=boto.ec2.autoscale.connect_to_region('eu-west-1')
con.get_all_launch_configurations()</code>
[LaunchConfiguration:removed, LaunchConfiguration:removed2, LaunchConfiguration:removed3]

**Comment by stevenbrichards** [comment]
@fvant what details are you looking for? The initial response of `get_all_launch_configurations()` is an array of all the launch configuration objects, each one should have all the following attributes available:
```
'associate_public_ip_address', 'block_device_mappings', 'classic_link_vpc_id', 'classic_link_vpc_security_groups', 'connection', 'created_time', 'delete', 'delete_on_termination', 'ebs_optimized', 'endElement', 'image_id', 'instance_monitoring', 'instance_profile_name', 'instance_type', 'iops', 'kernel_id', 'key_name', 'launch_configuration_arn', 'member', 'name', 'ramdisk_id', 'security_groups', 'spot_price', 'startElement', 'use_block_device_types', 'user_data', 'volume_type']
```
Such as:
```
>>> import boto.ec2.autoscale
>>> con=boto.ec2.autoscale.connect_to_region('us-west-1')
>>> con.get_all_launch_configurations()
[LaunchConfiguration:some_launch_configuration]
>>> con.get_all_launch_configurations()[0].image_id
u'ami-<redacted>'
```

**Comment by fvant** [comment]
All I get is a list of the LC names, nothing further, hence the reported
issue.

**Comment by stevenbrichards** [comment]
Can you do the following and paste the output?
```
import boto.ec2.autoscale 
con=boto.ec2.autoscale.connect_to_region('eu-west-1')
lc=con.get_all_launch_configurations()
dir(lc[0])
lc[0].image_id
```

**Comment by fvant** [comment]
Python 2.7.10 (default, Jun 30 2015, 15:30:23)
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import boto.ec2.autoscale
>>> con=boto.ec2.autoscale.connect_to_region('eu-west-1')
>>> lc=con.get_all_launch_configurations()
>>> dir(lc[0])
['__class__', '__delattr__', '__dict__', '__doc__', '__format__',
'__getattribute__', '__hash__', '__init__', '__module__', '__new__',
'__reduce__', '__reduce_ex__', '__repr__', '__setattr__',
'__sizeof__', '__str__', '__subclasshook__', '__weakref__',
'associate_public_ip_address', 'block_device_mappings',
'classic_link_vpc_id', 'classic_link_vpc_security_groups',
'connection', 'created_time', 'delete', 'delete_on_termination',
'ebs_optimized', 'endElement', 'image_id', 'instance_monitoring',
'instance_profile_name', 'instance_type', 'iops', 'kernel_id',
'key_name', 'launch_configuration_arn', 'member', 'name',
'ramdisk_id', 'security_groups', 'spot_price', 'startElement',
'use_block_device_types', 'user_data', 'volume_type']
>>> lc[0].image_id
u'ami-cb7de3bc'

**Comment by fvant** [comment]
I tried this similar version in boto3 and it workd just fine (giving me a
full dict)

import boto3
client = boto3.client('autoscaling')
res = client.describe_launch_configurations()
print res

**Comment by stevenbrichards** [comment]
In your above post where you did :
> >> lc[0].image_id
And got `u'ami-cb7de3bc'` show it is working, What `lc` is is a list of `LaunchConfiguration` objects, the output of `dir(lc[0])` shows all the attributes for a `LaunchConfiguration` object. Are you expecting a `dict` with `get_all_launch_configurations()`?

**Comment by fvant** [comment]
yes, the dict is also expected by for example Ansible (https://github.com/ansible/ansible-modules-core/issues/1655)

**Comment by stevenbrichards** [comment]
Ahhh it must've been a breaking change to the data type then. All the data is still there, just in a different representation. I can't give any insight as to why it was changed.

**Comment by fvant** [comment]
thing is, on other machines that also have 2.38.0 and it gives back a dict and all is fine. Just looks like mine (and other collegues) boto versions are different (even though we all just pip install boto).

**Comment by stevenbrichards** [comment]
Perhaps boto started calling a different method? (`describe_launch_configurations`) because `get_all_launch_configurations` has always returned a list since 2009. https://github.com/boto/boto/commit/e0de6ec0c0a9daec64806ea0a322ae6d15eefc80

**Comment by fvant** [comment]
describe_launch_configurations  is only boto3

**Comment by tamarabartlett** [comment]
@fvant Is this still an issue?
